File: Unidad 3 While/Programa21U3.java
package com.mycompany.programa21u3;

import java.util.Scanner;

public class Programa21U3 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Ingrese la cantidad de numeros naturales: ");
        int cn = sc.nextInt();

        int i = 1;
        while (i <= cn) {
            System.out.print("Ingrese el numero " + i + ": ");
            int numero = sc.nextInt();

            if (numero > 0) {
                int cubo = numero * numero * numero;
                System.out.println("El cubo de " + numero + " es: " + cubo);
                i++;
            } else {
                System.out.println("Ingrese un numero natural positivo.");
            }
        }
    }
}

File: Unidad 3 While/Programa21U3.cs
using System;

namespace Programa21U3
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.Write("Ingrese la cantidad de numeros naturales: ");
            int cn = Convert.ToInt32(Console.ReadLine());

            int i = 1;
            while (i <= cn)
            {
                Console.Write("Ingrese el numero " + i + ": ");
                int numero = Convert.ToInt32(Console.ReadLine());

                if (numero > 0)
                {
                    int cubo = numero * numero * numero;
                    Console.WriteLine("El cubo de " + numero + " es: " + cubo);
                    i++;
                }
                else
                {
                    Console.WriteLine("Ingrese un numero natural positivo.");
                }
            }
        }
    }
}